presto-main/src/main/java/com/facebook/presto/operator/SettableLookupSourceSupplier.java

```diff
@@ -46,10 +46,12 @@ public SettableLookupSourceSupplier(List<Type> types)
         return types;
     }
 
+    @SuppressWarnings("RedundantCast")
     @Override
     public ListenableFuture<LookupSource> getLookupSource(OperatorContext operatorContext)
     {
-        return transform(lookupSourceFuture, (AsyncFunction<SharedLookupSource, LookupSource>) Futures::immediateFuture);
+        // The Eclipse compiler requires the explicit <LookupSource> cast.
+        return transform(lookupSourceFuture, (AsyncFunction<SharedLookupSource, LookupSource>) Futures::<LookupSource>immediateFuture);
```

Comment:
That bug seems to be fixed in 4.4.2. I see this in the dependency for plexus-compiler-eclipse:
```
<groupId>org.eclipse.tycho</groupId>
<artifactId>org.eclipse.jdt.core</artifactId>
<version>3.10.0.v20140604-1726</version>
```
Does this mean it is using a very old version? cc @jvanzyl